snap/snapcraft.yaml

```diff
@@ -5,8 +5,78 @@ summary: Snap for NVIDIA DCGM and DCGM exporter
 license: Apache-2.0
 contact: solutions-engineering@lists.canonical.com
 description: |
-  This snap includes NVIDIA DCGM and DCGM exporter to provide GPU monitoring via
-  Prometheus metrics. This snap does not include the dcgmproftester.
+  This snap includes NVIDIA DCGM and DCGM-exporter to monitor GPU via Prometheus metrics
+  A use can setup a Grafana dashboard to visualize the metrics.
+
+  The snap includes the following components:
+  - DCGM: Data Center GPU Manager
+  - DCGM-exporter: A Prometheus exporter for DCGM metrics
+  - NV-Hostengine: A service that communicates with the NVIDIA GPU driver to get the metrics
+
+  **Note**: The `DCGM-exporter` service is disabled by default because a user might not be interested
+  in collecting metrics as this is a DCGM snap, not the DCGM-exporter snap.
+
+  **Quick Start**
+  ---
+  Getting the NVIDIA GPU metrics would look like this:
+
+  ```
+  # Install the snap
+  sudo snap install dcgm
+
+  # Start the DCGM exporter service
+  sudo snap start dcgm.dcgm-exporter
+
+  # Get the metrics
+  curl localhost:9400/metrics
+  ```
+
+  **Config**
+  ---
+  The DCGM snap provides several configuration options through the `snap` command.
+
+  ```
+  # Get all the configuration options
+  sudo snap get dcgm
+
+  # Example of setting the NV-Hostengine port
+  sudo snap set dcgm nv-hostengine-port=5577
+
+  # Restart the NV-Hostengine service to apply the changes
+  sudo snap restart dcgm.nv-hostengine
+  ```
+
+  Available configs:
+  - `nv-hostengine-port`: The port on which the NV-Hostengine listens.
+    The default is `5555`.
+  - `dcgm-exporter-address`: The bind address which the DCGM exporter exposes for the metrics.
+    The default is `:9400`.
+  - `dcgm-exporter-metrics-file`: The name of the custom CSV metrics file can be provided (only the name, not the path).
+    The file should be placed in the `/var/snap/dcgm/common/` directory.
+    The default metrics are located in `/snap/dcgm/current/etc/dcgm-exporter/default-counters.csv`.
+    Please refer to the DCGM-exporter repository for more information on the CSV file format.
+
+  **Limitations**
+  ---
+  As the DCGM components are packed in the snap, some features and configurations are not available.
```

Comment:
this makes it sound like snaps are an inferior technology. How about just mentioning that we don't currently provide dcgmproftester to limit the size of the snap?